doc/Manuals/ug-sec_x_road_security_hardening.md

```diff
@@ -0,0 +1,165 @@
+
+**Example on Ubuntu**
+
+Create a new configuration `/etc/pam.d/xroad` with the following content:
+```shell
+auth        required          pam_tally2.so deny=3 even_deny_root unlock_time=900 file=/var/lib/xroad/tallylog
+@include    common-auth    
+account     required          pam_tally2.so
+@include    common-account
+password    required          pam_deny.so    
+session     required          pam_deny.so    
+```
+
+**Example on RHEL**
+
+On RHEL systems, the `/etc/pam.d/xroad` file ships with the installation package so you need to modify the existing file. Replace the `/etc/pam.d/xroad` contents with the following:
+```shell
+#%PAM-1.0
+auth       required     pam_tally2.so deny=3 even_deny_root unlock_time=900 file=/var/lib/xroad/tallylog
+auth       required     pam_unix.so
+account    required     pam_tally2.so
+account    required     pam_unix.so
+password   required     pam_deny.so
+password   required     pam_warn.so
+session    required     pam_deny.so
+```
+
+### 2.2 Configuring password policies
+
+Configuring a password policy in your Admin UI authentication will provide an additional layer of defence against password guessing attacks, such as password spraying.
+
+User account passwords cannot be changed directly from the Admin UI, therefore the password policy must be configured on operating system user account level. 
+The method of adding a new password policy varies significantly depending on your operating system, existing PAM configuration and authentication protocol.
+
+For instruction on how to add password policies, please refer to your operating system's official documentation or customer support.
+
+#### 2.2.1 Considerations and risks
+
+In a strong password, it is advisable to have at least 16 characters at minimum. You can also add complexity requirements, such as numbers and special characters, but these requirements can make the passwords more difficult for users to remember. Further additional measures could be to add commonly known passwords into a blocklist.
+
+### 2.3 Ensuring User Account Security
+
+Users of the web application are created by creating operating-system-level users. This means that a user can access the web application and the underlying operating system with the same credentials. 
+Therefore, if user accounts in the web application were compromised, the attacker could use those credentials to log into the server via SSH if credential-based logging in is not disabled.
+
+To harden the user account security, make sure that users are not allowed to access the server via SSH by default. The users needing SSH access are granted those rights separately.
+
+Create a user group in which users are allowed to connect to the server via SSH while all other users are denied. Add the following line to `/etc/ssh/sshd_config`:
```

Comment:
I think it would be good to also add a reminder here to make sure that they add their user to the group (similar to the SSH key reminder below).